hoot-core-test/src/test/cpp/hoot/core/io/OsmApiChangesetTest.cpp

```diff
@@ -196,6 +197,90 @@ class OsmApiChangesetTest : public HootTestFixture
     QString expectedError = FileUtils::readFully("test-files/io/OsmChangesetElementTest/ChangesetErrorFixErrors.osc");
     HOOT_STR_EQUALS(expectedError, error);
   }
+
+  void runXmlChangesetFailureMatchesTest()
+  {
+    XmlChangeset changeset;
+
+    bool found = false;
+    long id = 0;
+    long id2 = 0;
+    ElementType::Type type = ElementType::Unknown;
+    ElementType::Type type2 = ElementType::Unknown;
+    QString hint;
+
+    //  Test Placeholder failure with node in way
+    id = 0;
+    id2 = 0;
+    type = ElementType::Unknown;
+    type2 = ElementType::Unknown;
+    hint = "Placeholder node not found for reference -145213 in way -5687";
+    found = changeset.matchesPlaceholderFailure(hint, id, type, id2, type2);
+    CPPUNIT_ASSERT_EQUAL(true, found);
+    CPPUNIT_ASSERT_EQUAL(-145213L, id);
+    CPPUNIT_ASSERT_EQUAL(ElementType::Node, type);
+    CPPUNIT_ASSERT_EQUAL(-5687L, id2);
+    CPPUNIT_ASSERT_EQUAL(ElementType::Way, type2);
```

Comment:
Each of these tests just test the parsing of two types of OSM API error messages that are dealt with differently than the default "bisect" strategy.  Push a string in, parse out the element type(s) and ID(s).